### DISCUSSION TITLE
lto-wrapper: fatal error: /usr/bin/avr-gcc returned 1 exit status compilation terminated.

### DISCUSSION THREAD

**Discussion by Gudzys** [olikraus/u8g2#2034] [Q&A]
Trying to compile the u8x8 sketch in the u8g2 library and get lto-wrapper: fatal error: /usr/bin/avr-gcc returned 1 exit status
compilation terminated. Researched it and installed board manager 1.6.21; installed
http://downloads.arduino.cc/packages/package_avr_7.3.0_index.json; and installed arduino-Makefile (yay -S arduino-mk ).  Still get the error.

**Comment by olikraus** [comment]
U8x8 is part of u8g2. U8g2 can be installed from Arduino IDE

**Comment by Gudzys** [comment]
It's installed but the full buffer u8g2 programs said there wasn't enough RAM for the local variables. Some of the u8x8 programs compile fine; this one did not. Is there a way to get around the memory limitation of the full buffer u8g2 programs? Thanks.

**Reply by olikraus** [reply]
> Is there a way to get around the memory limitation of the full buffer u8g2 programs?

Use page buffer mode of u8g2 or get a uC with more RAM.

**Reply by olikraus** [reply]
> this one did not. 

You mean, one of the u8x8 examples does not compile? Can you name the failing u8x8 example?

**Comment by Gudzys** [comment]
GraphicsTest gives me /usr/bin/avr-ld: error: lto-wrapper failed

Also get that on some other programs outside of u8g2. Did the solutions mentioned but they did not work. Am running Arduino IDE on Manjaro Linux: that may be the problem, not your program. Some have said this is a linux bug.

Thanks,

Brent

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Reply by olikraus** [reply]
u8x8 GraphicsTest compiles nicely for Arduino Uno with Arduino 1.8.4 on my system (Ubuntu Linux).

**Reply by Gudzys** [reply]
Appears to be an Arch/Manjaro linux bug.

**Reply by olikraus** [reply]
Oh, ok
